conan/api/model.py

```diff
@@ -85,3 +91,66 @@ def prefs(ref, recipe_bundle):
 
     def serialize(self):
         return self.recipes
+
+
+class ListPattern:
```

Comment:
This was an ``conan.internal`` import, so I had to move it and make it public ,as it was being used by commands, and it is an API public model. I also renamed it from SelectPattern->ListPattern